### PR TITLE
Feat: support any Terraform 1.x.x versions

### PR DESCRIPTION
**Pull request by TomerHeber**
No need to test each version.
Instead, test the latest. Create a "fork", only when there's an incompatibility issue.
So far 1.x.x are the same.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
         id: test-suite
         env:
           TF_VERSION: ${{ matrix.terraform_version }}
-        uses: actions/github-script@v4
+        uses: actions/github-script@v6
```
Comment (TomerHeber) [inline .github/workflows/ci.yml:82]: Deprecation warning.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -90,7 +90,7 @@ jobs:
 
       - name: Test
         run: |
-          go test -v -run ^TestTerraform${{ steps.test-suite.outputs.result }}$
+          go test -v -run ^TestTerraform${{ steps.test-suite.outputs.result }}
```
Comment (TomerHeber) [inline .github/workflows/ci.yml:93]: This did run all test cases. Removed the $

File: internal/file/file.go [hunk 3]
```diff
@@ -38,7 +37,7 @@ func ReplaceWithTerratagFile(path string, textContent string, rename bool) error
 
 func CreateFile(path string, textContent string) error {
 	log.Print("[INFO] Creating file ", path)
-	return ioutil.WriteFile(path, []byte(textContent), 0644)
```
Comment (TomerHeber) [inline internal/file/file.go:41]: ioutil is deprecated.

File: test/tests/aws_azure/expected/main.terratag.tf [hunk 4]
```diff
@@ -45,3 +45,4 @@ resource "aws_s3_bucket" "should_have_tags" {
 locals {
   terratag_added_main = {"env0_environment_id"="40907eff-cf7c-419a-8694-e1c6bf1d1168","env0_project_id"="43fd4ff1-8d37-4d9d-ac97-295bd850bf94"}
 }
+
```
Comment (TomerHeber) [inline test/tests/aws_azure/expected/main.terratag.tf:48]: test was failing (was not detected because the test was not running).

**Comment by TomerHeber** [comment]
@yaronya - after this is reviewed. Need to fix the checks. Otherwise this cannot be pushed.

**Review by yaronya — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -37,7 +37,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        terraform_version: ['11','12','13','14','15','1.0','1.1','1.2','1.3']
+        terraform_version: ['11','12','13','14','latest']
```
Comment (yaronya) [inline .github/workflows/ci.yml:40]: You removed the support for `0.15.x` is that on purpose?

**Review by yaronya — APPROVED** [review]

**Pull request merged**